# **(Data Collection Notebook)**

Part of CRISP-DM Data Understanding

## Objectives

* Fetch data from Kaggle and save as raw data
* Inspect the data and save it under outputs/datasets

## Inputs

* kaggle.JSON file - authentication token to access Kaggle dataset

## Outputs

* Generate dataset - 'outputs/datasets/raw/house-price-20211124T154130Z-001/house-price/house_prices_records.csv'
* Generate dataset - 'outputs/datasets/raw/house-price-20211124T154130Z-001/houseprice/inherited_houses.csv'

## Additional Comments

* The house_prices_records.csv is the data used to build the machine learning models.
* The inherited_houses.csv is the data with the inherited houses which the client would like us to predict.


---

# Change working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [16]:
%pip install -r /workspace/ames-heritage-housing/requirements.txt


[notice] A new release of pip is available: 23.0.1 -> 24.0
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [17]:
import os
current_dir = os.getcwd()
current_dir

'/workspace/ames-heritage-housing'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [18]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [19]:
current_dir = os.getcwd()
current_dir

'/workspace'

---

# Fetch the data from Kaggle

First, Install the Kaggle package to fetch data

In [20]:
! pip install kaggle==1.5.12


[notice] A new release of pip is available: 23.0.1 -> 24.0
[notice] To update, run: python3.8 -m pip install --upgrade pip


If JSON is already installed, this step can be skipped.

* Use the kaggle.json token to be able to authenticate and access the data:
* Run the cell below so that the token is recognised

In [21]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = os.getcwd()
! chmod 600 kaggle.json

chmod: cannot access 'kaggle.json': No such file or directory


The following dataset has been provided by the client:
* [Housing Prices Data](/https://www.kaggle.com/datasets/codeinstitute/housing-prices-data)
* Download the dataset and set to a destination folder

Get the dataset path from the Kaggle url.

In [22]:
KaggleDatasetPath = "codeinstitute/housing-prices-data"
DestinationFolder = "inputs/datasets/raw"
! kaggle datasets download -d {KaggleDatasetPath} -p {DestinationFolder}

Traceback (most recent call last):
  File "/home/gitpod/.pyenv/versions/3.8.18/bin/kaggle", line 33, in <module>
    sys.exit(load_entry_point('kaggle==1.5.12', 'console_scripts', 'kaggle')())
  File "/home/gitpod/.pyenv/versions/3.8.18/bin/kaggle", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/workspace/.pyenv_mirror/fakeroot/versions/3.8.18/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/workspace/.pyenv_mirror/fakeroot/versions/3.8.18/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>",

Unzip the downloaded file, delete the zip file and delete kaggle.json file

In [23]:
! unzip {DestinationFolder}/*.zip -d {DestinationFolder} \
    && rm {DestinationFolder}/*.zip \
    && rm kaggle.json

unzip:  cannot find or open inputs/datasets/raw/*.zip, inputs/datasets/raw/*.zip.zip or inputs/datasets/raw/*.zip.ZIP.

No zipfiles found.


---

# Load and Inspect Kaggle data

In this section we will load and inspect the inherited_houses and house_prices_records data:

### Inherited Houses

Load inherited_houses data:

In [24]:
import pandas as pd
df_inherited_houses = pd.read_csv(f"inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/inherited_houses.csv")
df_inherited_houses.head()

FileNotFoundError: [Errno 2] No such file or directory: 'inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/inherited_houses.csv'

Dataframe summary: 
* Inspect inherited_houses data

In [ ]:
df_inherited_houses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4 entries, 0 to 3
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   1stFlrSF       4 non-null      int64  
 1   2ndFlrSF       4 non-null      int64  
 2   BedroomAbvGr   4 non-null      int64  
 3   BsmtExposure   4 non-null      object 
 4   BsmtFinSF1     4 non-null      float64
 5   BsmtFinType1   4 non-null      object 
 6   BsmtUnfSF      4 non-null      float64
 7   EnclosedPorch  4 non-null      int64  
 8   GarageArea     4 non-null      float64
 9   GarageFinish   4 non-null      object 
 10  GarageYrBlt    4 non-null      float64
 11  GrLivArea      4 non-null      int64  
 12  KitchenQual    4 non-null      object 
 13  LotArea        4 non-null      int64  
 14  LotFrontage    4 non-null      float64
 15  MasVnrArea     4 non-null      float64
 16  OpenPorchSF    4 non-null      int64  
 17  OverallCond    4 non-null      int64  
 18  OverallQual   

Check for duplicated data:

In [ ]:
df_inherited_houses[df_inherited_houses.duplicated(subset=None)]

Empty DataFrame
Columns: [1stFlrSF, 2ndFlrSF, BedroomAbvGr, BsmtExposure, BsmtFinSF1, BsmtFinType1, BsmtUnfSF, EnclosedPorch, GarageArea, GarageFinish, GarageYrBlt, GrLivArea, KitchenQual, LotArea, LotFrontage, MasVnrArea, OpenPorchSF, OverallCond, OverallQual, TotalBsmtSF, WoodDeckSF, YearBuilt, YearRemodAdd]
Index: []

[0 rows x 23 columns]

No duplicated data

#### Findings

* The dataset contains 4 entries of data and 23 columns.
* There is a mix of the following data types: int64, float64 and object.

### House Prices

Load house_prices_records data:

In [ ]:
import pandas as pd
df_house_prices = pd.read_csv(f"inputs/datasets/raw/house-price-20211124T154130Z-001/house-price/house_prices_records.csv")
df_house_prices.head()

1stFlrSF  2ndFlrSF  BedroomAbvGr BsmtExposure  BsmtFinSF1 BsmtFinType1  \
0       856     854.0           3.0           No         706          GLQ   
1      1262       0.0           3.0           Gd         978          ALQ   
2       920     866.0           3.0           Mn         486          GLQ   
3       961       NaN           NaN           No         216          ALQ   
4      1145       NaN           4.0           Av         655          GLQ   

   BsmtUnfSF  EnclosedPorch  GarageArea GarageFinish  ...  LotFrontage  \
0        150            0.0         548          RFn  ...         65.0   
1        284            NaN         460          RFn  ...         80.0   
2        434            0.0         608          RFn  ...         68.0   
3        540            NaN         642          Unf  ...         60.0   
4        490            0.0         836          RFn  ...         84.0   

   MasVnrArea OpenPorchSF  OverallCond  OverallQual  TotalBsmtSF  WoodDeckSF  \
0       196.0          61            5            7          856         0.0   
1         0.0           0            8            6         1262         NaN   
2       162.0          42            5            7          920         NaN   
3         0.0          35            5            7          756         NaN   
4       350.0          84            5            8         1145         NaN   

   YearBuilt  YearRemodAdd  SalePrice  
0       2003          2003     208500  
1       1976          1976     181500  
2       2001          2002     223500  
3       1915          1970     140000  
4       2000          2000     250000  

[5 rows x 24 columns]

Dataframe summary:
* Inspect house_prices_records data

In [ ]:
df_house_prices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 24 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   1stFlrSF       1460 non-null   int64  
 1   2ndFlrSF       1374 non-null   float64
 2   BedroomAbvGr   1361 non-null   float64
 3   BsmtExposure   1460 non-null   object 
 4   BsmtFinSF1     1460 non-null   int64  
 5   BsmtFinType1   1346 non-null   object 
 6   BsmtUnfSF      1460 non-null   int64  
 7   EnclosedPorch  136 non-null    float64
 8   GarageArea     1460 non-null   int64  
 9   GarageFinish   1298 non-null   object 
 10  GarageYrBlt    1379 non-null   float64
 11  GrLivArea      1460 non-null   int64  
 12  KitchenQual    1460 non-null   object 
 13  LotArea        1460 non-null   int64  
 14  LotFrontage    1201 non-null   float64
 15  MasVnrArea     1452 non-null   float64
 16  OpenPorchSF    1460 non-null   int64  
 17  OverallCond    1460 non-null   int64  
 18  OverallQ

Check for duplicated data:

In [ ]:
df_house_prices[df_house_prices.duplicated(subset=None)]

Empty DataFrame
Columns: [1stFlrSF, 2ndFlrSF, BedroomAbvGr, BsmtExposure, BsmtFinSF1, BsmtFinType1, BsmtUnfSF, EnclosedPorch, GarageArea, GarageFinish, GarageYrBlt, GrLivArea, KitchenQual, LotArea, LotFrontage, MasVnrArea, OpenPorchSF, OverallCond, OverallQual, TotalBsmtSF, WoodDeckSF, YearBuilt, YearRemodAdd, SalePrice]
Index: []

[0 rows x 24 columns]

No duplicated data

#### Findings

* The dataset contains 1460 entries of data and 24 columns (there is one more column in this dataset - SalePrice - due to house prices already available and will be used as part of the machine learning)
* There is a mix of the following data types: int64, float64, object.

---

# Push files to Repo

Create outputs/datasets/collection folder

In [ ]:
import os
try:
  os.makedirs(name='outputs/datasets/collection')
except Exception as e:
  print(e)

df_inherited_houses.to_csv(f"outputs/datasets/collection/inherited_houses.csv", index=False)
df_house_prices.to_csv(f"outputs/datasets/collection/house_prices_records.csv", index=False)


[Errno 17] File exists: 'outputs/datasets/collection'
